### ISSUE TITLE
Enumearable.join() does not reference outer table

### ISSUE THREAD

**Issue by Kevok93** [bridgedotnet/Bridge#3977]
Full cs in deck, below is minimized

### Steps To Reproduce

https://deck.net/354671f3325fe6b0fdc7a42cf0bbc1b5

```csharp
    struct stringHolder { string _val; }
        //star join t1*t2 where 1==1
        try {
            System.Linq.Enumerable.Join(
                inner: (stringHolder[]) table1,
                outer: (stringHolder[]) table2,
                innerKeySelector: row1 => true,
                outerKeySelector: row2 => true,
                resultSelector: (stringHolder t1Val, stringHolder t2Val) => {
                    if (!t2Val.HasOwnProperty("_val")) //does not throw
                        throw new Exception ("Still broken!");
                    
                    if (!t1Val.HasOwnProperty("_val")) //throws
                        throw new Exception ("Still broken!");
                    
                    System.Console.WriteLine($"{t1Val._val} {t2Val._val}!");
                    return true;
                }
            )
```

### Expected Result

```js
                   table1.join(table2, function (row2) {
                        return true;
                    }, function (row1) {
                        return true;
                    }, function (t1Val, t2Val) {
                        if (!t2Val.hasOwnProperty("_val")) {
                            throw new System.Exception("Still broken!");
                        }

                        if (!t1Val.hasOwnProperty("_val")) {
                            throw new System.Exception("Still broken!");
                        }

                        System.Console.WriteLine(System.String.format("{0} {1}!", t1Val._val, t2Val._val));
                        return true;
                    }
                  /*OR*/
                   System.Linq.Enumerable.join(table1, table2, function (row2) {
                        return true;
                    }, function (row1) {
                        return true;
                    }, function (t1Val, t2Val) {
                        if (!t2Val.hasOwnProperty("_val")) {
                            throw new System.Exception("Still broken!");
                        }

                        if (!t1Val.hasOwnProperty("_val")) {
                            throw new System.Exception("Still broken!");
                        }

                        System.Console.WriteLine(System.String.format("{0} {1}!", t1Val._val, t2Val._val));
                        return true;
                    }
```

### Actual Result

```js
                System.Linq.Enumerable.from(System.Linq.Enumerable).join(table1, function (row2) {
                        return true;
                    }, function (row1) {
                        return true;
                    }, function (t1Val, t2Val) {
                        if (!t2Val.hasOwnProperty("_val")) {
                            throw new System.Exception("Still broken!");
                        }

                        if (!t1Val.hasOwnProperty("_val")) {
                            throw new System.Exception("Still broken!");
                        }

                        System.Console.WriteLine(System.String.format("{0} {1}!", t1Val._val, t2Val._val));
                        return true;
                    }

```

**Comment by geoffreymcgill** [comment]
Thanks for the report. A fix for this defect has been created. After additional review and testing, the fix will be included in the upcoming Bridge 17.8.0 release.

**Comment by fabriciomurta** [comment]
The fix looks good and has been merged upstream!
